onionshare_gui/onionshare_gui.py

```diff
@@ -194,8 +199,8 @@ def update_mode_switcher(self):
             self.share_mode_button.setStyleSheet(self.common.css['mode_switcher_selected_style'])
             self.receive_mode_button.setStyleSheet(self.common.css['mode_switcher_unselected_style'])
 
-            self.share_mode.show()
             self.receive_mode.hide()
+            self.share_mode.show()
```

Comment:
I don't know why, but undoing this change whereby `self.share_mode.show()` occurs first (before `self.receive_mode.hide()` is called), is enough to fix the weird download history widget from being resized badly (and the indicator icons moving to the left), per my earlier review. 

It does seem to cause a slight vertical stretch in share mode when this change is reverted. but I'm not sure that's a big deal, share mode is always 'bigger' anyway due to the FileList and associated widgets.

If you were to revert this change, and re-add the `upload.close()` as per the other comment, and fix the conflicts, I think this could be merge-able (and then #800 could be merged and we could move on to fixing the rest of the tests)